TCPSocketConnectionTimeoutIT error running verify

When I run  

```
MANTA_URL=https://manta.garage.local mvn -Dmanta.user=java_test 
-Dmanta.key_id=SHA256:vOlOMddoqBTCaCrO1BQTc0D1Zi+DzfvxHlZyg9JYvxY
-Dmanta.key_path=/home/russell/dev/joyent/repos/java-manta/java_id_rsa  -e verify
```
 I get

```
Cannot instantiate class com.joyent.manta.http.TCPSocketConnectionTimeoutIT
	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:673)
	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:535)
	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:280)
	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:245)
	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1124)
	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:954)
	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:832)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:956)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)

```

The "dump" file has a slightly fuller stack trace
```
[root@52165b86-a1d3-6a0e-96c0-cea209198845 /var/tmp/tmp-manta-nightly-sdk]# cat java-manta-it/target/failsafe-reports/2019-05-20T20-35-18_863-jvmRun1.dump 
# Created on 2019-05-20T20:35:21.491
org.testng.TestNGException: 
Cannot instantiate class com.joyent.manta.http.TCPSocketConnectionTimeoutIT
        at org.testng.internal.ObjectFactoryImpl.newInstance(ObjectFactoryImpl.java:30)
        at org.testng.internal.ClassHelper.createInstance1(ClassHelper.java:423)
        at org.testng.internal.ClassHelper.createInstance(ClassHelper.java:336)
        at org.testng.internal.ClassImpl.getDefaultInstance(ClassImpl.java:125)
        at org.testng.internal.ClassImpl.getInstances(ClassImpl.java:190)
        at org.testng.TestClass.getInstances(TestClass.java:95)
        at org.testng.TestClass.initTestClassesAndInstances(TestClass.java:81)
        at org.testng.TestClass.init(TestClass.java:73)
        at org.testng.TestClass.<init>(TestClass.java:38)
        at org.testng.TestRunner.initMethods(TestRunner.java:389)
        at org.testng.TestRunner.init(TestRunner.java:271)
        at org.testng.TestRunner.init(TestRunner.java:241)
        at org.testng.TestRunner.<init>(TestRunner.java:167)
        at org.testng.SuiteRunner$DefaultTestRunnerFactory.newTestRunner(SuiteRunner.java:663)
        at org.testng.SuiteRunner.init(SuiteRunner.java:260)
        at org.testng.SuiteRunner.<init>(SuiteRunner.java:198)
        at org.testng.TestNG.createSuiteRunner(TestNG.java:1295)
        at org.testng.TestNG.createSuiteRunners(TestNG.java:1273)
        at org.testng.TestNG.runSuitesLocally(TestNG.java:1128)
        at org.testng.TestNG.runSuites(TestNG.java:1049)
        at org.testng.TestNG.run(TestNG.java:1017)
        at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
        at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
        at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
        at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:373)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:334)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:119)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:407)
Caused by: java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at org.testng.internal.ObjectFactoryImpl.newInstance(ObjectFactoryImpl.java:24)
        ... 27 more
Caused by: com.joyent.manta.exception.ConfigurationException: Given fingerprint c4:13:be:5c:fe:a7:53:ce:c2:03:53:2b:f7:47:14:e2 does not match expected key MD5:de:e7:73:9a:b0:91:31:3e:72:8d:9f:62:ca:58:a2:ec SHA256:2XCGHtsnTro2MGqoVQHgXSS/ai12yQHrGFrN12YUPpA
Exception Context:
        [1:mantaSdkVersion=3.4.1-SNAPSHOT]
---------------------------------
        at com.joyent.manta.config.KeyPairFactory.createKeyPair(KeyPairFactory.java:85)
        at com.joyent.manta.config.AuthAwareConfigContext.doLoad(AuthAwareConfigContext.java:107)
        at com.joyent.manta.config.AuthAwareConfigContext.reload(AuthAwareConfigContext.java:88)
        at com.joyent.manta.config.AuthAwareConfigContext.<init>(AuthAwareConfigContext.java:64)
        at com.joyent.manta.http.TCPSocketConnectionTimeoutIT.<init>(TCPSocketConnectionTimeoutIT.java:37)
        ... 32 more
```

> The "dump" file has a slightly fuller stack trace
> 
> ```
> [root@52165b86-a1d3-6a0e-96c0-cea209198845 /var/tmp/tmp-manta-nightly-sdk]# cat java-manta-it/target/failsafe-reports/2019-05-20T20-35-18_863-jvmRun1.dump 
> # Created on 2019-05-20T20:35:21.491
> org.testng.TestNGException: 
> Cannot instantiate class com.joyent.manta.http.TCPSocketConnectionTimeoutIT
>         at org.testng.internal.ObjectFactoryImpl.newInstance(ObjectFactoryImpl.java:30)
>         at org.testng.internal.ClassHelper.createInstance1(ClassHelper.java:423)
>         at org.testng.internal.ClassHelper.createInstance(ClassHelper.java:336)
>         at org.testng.internal.ClassImpl.getDefaultInstance(ClassImpl.java:125)
>         at org.testng.internal.ClassImpl.getInstances(ClassImpl.java:190)
>         at org.testng.TestClass.getInstances(TestClass.java:95)
>         at org.testng.TestClass.initTestClassesAndInstances(TestClass.java:81)
>         at org.testng.TestClass.init(TestClass.java:73)
>         at org.testng.TestClass.<init>(TestClass.java:38)
>         at org.testng.TestRunner.initMethods(TestRunner.java:389)
>         at org.testng.TestRunner.init(TestRunner.java:271)
>         at org.testng.TestRunner.init(TestRunner.java:241)
>         at org.testng.TestRunner.<init>(TestRunner.java:167)
>         at org.testng.SuiteRunner$DefaultTestRunnerFactory.newTestRunner(SuiteRunner.java:663)
>         at org.testng.SuiteRunner.init(SuiteRunner.java:260)
>         at org.testng.SuiteRunner.<init>(SuiteRunner.java:198)
>         at org.testng.TestNG.createSuiteRunner(TestNG.java:1295)
>         at org.testng.TestNG.createSuiteRunners(TestNG.java:1273)
>         at org.testng.TestNG.runSuitesLocally(TestNG.java:1128)
>         at org.testng.TestNG.runSuites(TestNG.java:1049)
>         at org.testng.TestNG.run(TestNG.java:1017)
>         at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
>         at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
>         at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
>         at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:373)
>         at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:334)
>         at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:119)
>         at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:407)
> Caused by: java.lang.reflect.InvocationTargetException
>         at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
>         at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
>         at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
>         at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
>         at org.testng.internal.ObjectFactoryImpl.newInstance(ObjectFactoryImpl.java:24)
>         ... 27 more
> Caused by: com.joyent.manta.exception.ConfigurationException: Given fingerprint c4:13:be:5c:fe:a7:53:ce:c2:03:53:2b:f7:47:14:e2 does not match expected key MD5:de:e7:73:9a:b0:91:31:3e:72:8d:9f:62:ca:58:a2:ec SHA256:2XCGHtsnTro2MGqoVQHgXSS/ai12yQHrGFrN12YUPpA
> Exception Context:
>         [1:mantaSdkVersion=3.4.1-SNAPSHOT]
> ---------------------------------
>         at com.joyent.manta.config.KeyPairFactory.createKeyPair(KeyPairFactory.java:85)
>         at com.joyent.manta.config.AuthAwareConfigContext.doLoad(AuthAwareConfigContext.java:107)
>         at com.joyent.manta.config.AuthAwareConfigContext.reload(AuthAwareConfigContext.java:88)
>         at com.joyent.manta.config.AuthAwareConfigContext.<init>(AuthAwareConfigContext.java:64)
>         at com.joyent.manta.http.TCPSocketConnectionTimeoutIT.<init>(TCPSocketConnectionTimeoutIT.java:37)
>         ... 32 more
> ```

Chris did you perform this test after incorporating the changes introduced by #517 ?

In fact, this is odd:-

```
Caused by: com.joyent.manta.exception.ConfigurationException: Given fingerprint SHA256:N5Qi3xOV9WweY0i1kohPcdRrzAMwIQ6SxIST8G4bJTo does not match expected key 
MD5:a2:65:bf:30:3f:d5:ab:17:c2:11:61:b2:3c:5b:20:0e SHA256:vOlOMddoqBTCaCrO1BQTc0D1Zi+DzfvxHlZyg9JYvxY
Exception Context:
```
when the "given fingerprint" is 

`-Dmanta.key_id=SHA256:vOlOMddoqBTCaCrO1BQTc0D1Zi+DzfvxHlZyg9JYvxY` That other fingerprint is from the environment variables. Am I to unset the env vars first? Do env vars trump -D type system properties?

--- a/java-manta-it/src/test/java/com/joyent/manta/http/TCPSocketConnectionTimeoutIT.java
+++ b/java-manta-it/src/test/java/com/joyent/manta/http/TCPSocketConnectionTimeoutIT.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2018, Joyent, Inc. All rights reserved.
+ * Copyright (c) 2018-2019, Joyent, Inc. All rights reserved.
  *
  * This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
@@ -17,6 +17,7 @@ import com.joyent.manta.config.StandardConfigContext;
 import com.joyent.manta.exception.MantaIOException;
 import org.apache.commons.lang3.exception.ExceptionUtils;
 import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
 import org.testng.annotations.AfterClass;
 import org.testng.annotations.Test;
 
@@ -25,19 +26,50 @@ import java.time.Duration;
 import java.time.Instant;
 
 /**
- * Integration tests for verifying the connection timeout behaviour for an invalid connection.
+ * Integration tests for verifying the connection timeout behaviour for an invalid connection,
+ * implies having an invalid value for MANTA_URL.
  * This Integration test is written for the sole purpose of verifying Issue #441.
  *
  * @author <a href="https://github.com/dekobon">Ashwin A Nair</a>
  */
-
 @Test
 public class TCPSocketConnectionTimeoutIT {
-    // Let TestNG configuration take precedence over environment variables
-    private AuthAwareConfigContext authConfig = new AuthAwareConfigContext(
-            new IntegrationTestConfigContext());
+    private MantaClient mantaClient;
 
-    private ChainedConfigContext config = new ChainedConfigContext(
+    private String testPathPrefix;
+
+    private ConfigContext config;
+
+    public TCPSocketConnectionTimeoutIT() {
+        // Let TestNG configuration take precedence over environment variables
+
+        config = new IntegrationTestConfigContext();
+        mantaClient = new MantaClient(config);
+        testPathPrefix = IntegrationTestConfigContext.generateBasePath(config, this.getClass().getSimpleName());
+    }
+
+    @BeforeClass
+    public void beforeClass() throws IOException {
+        mantaClient.putDirectory(testPathPrefix, true);
+    }
+
+    @AfterClass
+    public void cleanup() throws IOException {
+        IntegrationTestConfigContext.cleanupTestDirectory(mantaClient, testPathPrefix);
+    }
+
+    /**
+     * Test that verifies that the connection timeout setting
+     * {@link ConfigContext#getTimeout()} is applied and is actually working.
+     *
+     * Note: this test will not work correctly when run through a proxy server.
+     */
+    public void verifyConnectionTimeoutSettingWorks() throws IOException {
+        final AuthAwareConfigContext authConfig = new AuthAwareConfigContext(
+                new IntegrationTestConfigContext());
+
+        final ConfigContext badConfig = new ChainedConfigContext(
+                config,
                 new StandardConfigContext()
                         .setRetries(0)
                         .setTimeout(1000)
@@ -51,20 +83,10 @@ public class TCPSocketConnectionTimeoutIT {
                 new DefaultsConfigContext()
         );
 
-    /**
-     * Test that verifies that the connection timeout setting
-     * {@link ConfigContext#getTimeout()} is applied and is actually working.
-     *
-     * Note: this test will not work correctly when run through a proxy server.
-     */
-    public void verifyConnectionTimeoutSettingWorks() throws IOException {
-        String testPathPrefix = IntegrationTestConfigContext.generateBasePath(
-                config, this.getClass().getSimpleName());
-
         Instant start = null;
         Instant stop;
 
-        try (MantaClient client = new MantaClient(config)) {
+        try (MantaClient client = new MantaClient(badConfig)){
             start = Instant.now();
             client.head(testPathPrefix);
         } catch (MantaIOException e) {
@@ -81,15 +103,10 @@ public class TCPSocketConnectionTimeoutIT {
         Duration totalTime = Duration.between(start, stop);
 
         if (totalTime.getSeconds() != 1) {
-            String msg = String.format("Expected total connection time duration "
+            final String msg = String.format("Expected total connection time duration "
                     + "to be 1 second. Actually [%d] ms",
                     totalTime.toMillis());
             Assert.fail(msg);
         }
-    }
-
-    @AfterClass
-    public void afterClass() throws Exception {
-        authConfig.close();
     }
 }

